src/stores/CheckoutController.ts

```diff
@@ -1,152 +1,165 @@
+      const { paymentMethods } = useCheckoutStore.getState();
 
-  const { user, auth } = useAccountStore.getState();
-  const { paymentMethods } = useCheckoutStore.getState();
+      if (paymentMethods) return paymentMethods; // already fetched payment methods
 
-  if (paymentMethods) return paymentMethods; // already fetched payment methods
-  if (!cleengId) throw new Error('cleengId is not configured');
-  if (!user || !auth) throw new Error('user is not logged in');
+      const response = await checkoutService.getPaymentMethods(sandbox, jwt);
 
-  const response = await checkoutService.getPaymentMethods(cleengSandbox, auth.jwt);
+      if (response.errors.length > 0) throw new Error(response.errors[0]);
 
-  if (response.errors.length > 0) throw new Error(response.errors[0]);
+      useCheckoutStore.getState().setPaymentMethods(response.responseData?.paymentMethods);
 
-  useCheckoutStore.getState().setPaymentMethods(response.responseData?.paymentMethods);
-
-  return response.responseData?.paymentMethods;
+      return response.responseData?.paymentMethods;
+    });
+  });
 };
 
-export const paymentWithoutDetails = async () => {
-  const { cleengSandbox, cleengId } = useConfigStore.getState().getCleengData();
-
-  const { user, auth } = useAccountStore.getState();
-  const { order } = useCheckoutStore.getState();
+export const paymentWithoutDetails = async (): Promise<unknown> => {
+  return await useAccount(async ({ auth: { jwt } }) => {
+    return await useService(async ({ checkoutService, sandbox, authProviderId }) => {
+      const { order } = useCheckoutStore.getState();
 
-  if (!order) throw new Error('No order created');
-  if (!cleengId) throw new Error('cleengId is not configured');
-  if (!user || !auth) throw new Error('user is not logged in');
+      if (!order) throw new Error('No order created');
+      if (!authProviderId) throw new Error('auth provider is not configured');
 
-  const response = await checkoutService.paymentWithoutDetails({ orderId: order.id }, cleengSandbox, auth.jwt);
+      const response = await checkoutService.paymentWithoutDetails({ orderId: order.id }, sandbox, jwt);
 
-  if (response.errors.length > 0) throw new Error(response.errors[0]);
-  if (response.responseData.rejectedReason) throw new Error(response.responseData.rejectedReason);
+      if (response.errors.length > 0) throw new Error(response.errors[0]);
+      if (response.responseData.rejectedReason) throw new Error(response.responseData.rejectedReason);
 
-  return response.responseData;
+      return response.responseData;
+    });
+  });
 };
 
-export const adyenPayment = async (paymentMethod: AdyenPaymentMethod) => {
-  const { cleengSandbox, cleengId } = useConfigStore.getState().getCleengData();
-  const { user, auth } = useAccountStore.getState();
-  const { order } = useCheckoutStore.getState();
+export const cardPayment = async (cardPaymentPayload: CardPaymentData): Promise<unknown> => {
```

Comment:
Instead of calling these adyen and card payment, I think directPostCardPayment and iFrameCardPayment will make it more understandable. Unless it's possible to combine these 2 into a single path that takes a param of card details or reference details.

Reply:
updated.